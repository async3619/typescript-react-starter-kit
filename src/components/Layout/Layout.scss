.root {
    min-height: 100vh;

    margin: 0;
    padding: 0;

    display: flex;
    flex-direction: column;

    > .content {
        flex: 1 1 auto;
    }
}
