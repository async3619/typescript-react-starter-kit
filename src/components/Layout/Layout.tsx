import React, { FunctionComponent } from "react";
import useStyles from "isomorphic-style-loader/useStyles";

import Header from "@components/Header";
import Navigation from "@components/Navigation";

// external-global styles must be imported in your JS.
import normalizeCss from "normalize.css";
import s from "./Layout.scss";

interface PropTypes {}

const Layout: FunctionComponent<PropTypes> = ({ children }) => {
    useStyles(normalizeCss, s);

    return (
        <div>
            <Navigation />
            <Header />
            {children}
        </div>
    );
};

export default Layout;
