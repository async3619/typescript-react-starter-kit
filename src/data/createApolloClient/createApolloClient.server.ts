import { ApolloClient } from "apollo-client";
import { from } from "apollo-link";
import { onError } from "apollo-link-error";
import { SchemaLink } from "apollo-link-schema";
import createCache from "./createCache";

export default function createApolloClient(schema: SchemaLink.Options, partialCacheDefaults: Object) {
    const cache = createCache();

    cache.writeData({
        data: partialCacheDefaults,
    });

    const link = from([
        onError(({ graphQLErrors, networkError }) => {
            if (graphQLErrors)
                graphQLErrors.map(({ message, locations, path }) =>
                    console.warn(`[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`),
                );
            if (networkError) console.warn(`[Network error]: ${networkError}`);
        }),
        new SchemaLink({ ...schema }),
    ]);

    return new ApolloClient({
        // @ts-ignore
        link,
        cache,
        ssrMode: true,
        queryDeduplication: true,
    });
}
