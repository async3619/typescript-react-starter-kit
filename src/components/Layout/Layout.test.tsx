/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from "react";
import renderer from "react-test-renderer";
import createApolloClient from "../../data/createApolloClient/createApolloClient.server";
import App from "../App";

import Layout from ".";

describe("Layout", () => {
    test("renders children correctly", () => {
        const client = createApolloClient({} as any, {} as any);

        const wrapper = renderer
            .create(
                <App
                    context={{
                        pathname: "",
                        query: {},
                    }}
                    insertCss={() => {}}
                    client={client}
                >
                    <Layout>
                        <div className="child" />
                    </Layout>
                </App>,
            )
            .toJSON();

        expect(wrapper).toMatchSnapshot();
    });
});
