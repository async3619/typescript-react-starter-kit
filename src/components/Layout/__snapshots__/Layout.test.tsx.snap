// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <div
      className="wrapper root"
    >
      <typescript-logo.svg
        className="logo"
      />
      <h2
        className="title"
      >
        Typescript React Starter Kit
      </h2>
      <div
        className="placeholder"
      />
      <ul
        className="menu"
      >
        <li>
          <a
            href="https://github.com/async3619/typescript-react-starter-kit"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="root"
  >
    <div
      className="root"
    >
      <h1
        className="title"
      >
        Typescript React Starter Kit
      </h1>
      <h2
        className="subtitle"
      >
        Easier setting up project with essential features
      </h2>
    </div>
    <div
      className="bg"
    >
      <div
        className="bgWrapper root"
      >
        <div
          className="logoWrapper"
        >
          <div
            className="reactWrapper"
          >
            <logo.svg
              className="logo"
            />
          </div>
          <typescript-logo.svg
            className="logo typescript"
          />
        </div>
      </div>
    </div>
  </div>
  <main
    className="content"
  >
    <div
      className="root"
    >
      <div
        className="child"
      />
    </div>
  </main>
  <div
    className="root"
  >
    <div
      className="root"
    >
      <span
        className="copyright"
      >
        Â© 2019 Sophia &lt;async3619@naver.com&gt;
      </span>
    </div>
  </div>
</div>
`;
